SquareLine/gen.py

```diff
@@ -0,0 +1,231 @@
+        SLB_FILE["color_depth"] = "16 sw"
+    else:
+        SLB_FILE["color_depth"] = "16"
+    SLB_FILE["short_description"] = manifest["short_description"]
+    SLB_FILE["long_description"] = manifest["long_description"]
+
+    out_slb_json = json.dumps(SLB_FILE, indent=4)
+    f = open(output + "/" + output_filename + ".slb", "w")
+    f.write(out_slb_json)
+    f.close()
+
+    print(f"  File {output_filename}.slb created.")
+
+
+# Process of the board generating
+def process_board(board_name, output, dir):
+    global placeholders
+    print(f"Processing board: {dir}")
+
+    path = dir + "/" + BOARD_GENFILE
+    if os.path.exists(path):
+        # Parse manifest file
+        f = open(path, "r")
+        manifest = json.loads(f.read())
+        f.close()
+
+        # Get string version for file names
+        file_ver = get_file_version(manifest)
+        output_filename = board_name + file_ver
+
+        check_json_key(manifest, "placeholders")
+        placeholders = manifest["placeholders"]
+
+        # Copy common files
+        copy_directory(COMMON_DIR, output + "/" + SQUARELINE_OUTDIR)
+        # Copy specific board files
+        for file in BOARD_SPECIFIC_FILES:
+            copy_file(dir + "/" + file, output + "/" + SQUARELINE_OUTDIR + "/" + file)
+        # Copy image
+        copy_file(dir + "/image.png", output + "/" + output_filename + ".png", 0)
+
+        create_slb_file(output, output_filename, manifest)
+        make_zip(output_filename, output)
+        remove_folder(output + "/" + SQUARELINE_OUTDIR)
+    else:
+        print_error(f"ERROR: File '{path}' is not exists!")
+        os._exit(1)
+
+# Print usage
+def print_help():
+    print("Not enough arguments! Usage:")
+    print(" gen.py out board")
+    print("\tout: path to the generated output")
+    print("\tboard (optional): generate only specific board")
+
+
+def main():
+    if len(sys.argv) >= 2:
+        output_folder = sys.argv[1] + "/" + OUT_DIR
+
+        if len(sys.argv) > 2:
+            # Process only one specific board
+            board_name = sys.argv[2]
+            board_path = BOARDS_DIR + board_name
+            output_folder += "/" + board_name
+            # Remove all in output directory
+            if os.path.exists(output_folder):
+                shutil.rmtree(output_folder)
+            # Process
+            process_board(board_name, output_folder, board_path)
+        else:
+            # Remove all in output directory
+            if os.path.exists(output_folder):
+                shutil.rmtree(output_folder)
+            #loop all boards
+            for dirname in os.listdir(BOARDS_DIR):
+                d = os.path.join(BOARDS_DIR, dirname)
+                # checking if it is a file
+                if os.path.isdir(d):
+                    process_board(dirname, output_folder + "/" + dirname, d)
+
+        print_ok(f"Generating done! Output folder: {sys.argv[1]}")
+    else:
+        print_help()
```

Comment:
Should probably also `raise SystemExit(1)` in this case so that the return code is not "success".

(although if you switch to `argparse`, it will handle this automatically)